rs/nervous_system/temporary/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+//! This should probably not be used outside of test.
+//!
+//! This is for "feature flags". That is,
+//!
+//!     use ic_nervous_system_temporary::Temporary;
+//!     use std::cell::RefCell;
+//!     use rand::Rng;
+//!
+//!     thread_local! {
+//!         static IS_FOO_ENABLED: RefCell<bool> = RefCell::new(rand::thread_rng().gen());
+//!     }
+//!
+//!     pub fn is_foo_enabled() -> bool {
+//!         IS_FOO_ENABLED.with(|ok| {
+//!             let ok = ok.borrow();
+//!             *ok
+//!         })
+//!     }
+//!
+//!     fn code_under_test() -> i64 {
+//!         if is_foo_enabled() {
+//!             42
+//!         } else {
+//!             99
+//!         }
+//!     }
+//!
+//!     #[cfg(test)]
+//!     fn temporarily_enable_foo() -> Temporary {
+//!         Temporary::new(&FOO, true)
+//!     }
+//!
+//!     #[cfg(test)]
+//!     fn temporarily_disable_foo() -> Temporary {
+//!         Temporary::new(&FOO, false)
+//!     }
+//!
+//!     // Then, in tests, you can see what happens depending on whether the
+//!     // feature is or disabled without having to have separate
+//!     // feature = "test" builds, like so:
+//!
+//!     #[test]
+//!     fn test_foo_enabled() {
+//!         let _restore_foo_on_drop = temporarily_enable_foo();
+//!
+//!         let result = code_under_test();
+//!
+//!         assert_eq!(result, 42);
+//!     }
+//!
+//!     #[test]
+//!     fn test_foo_disabled() {
+//!         let _restore_foo_on_drop = temporarily_disable_foo();
+//!
+//!         let result = code_under_test();
+//!
+//!         assert_eq!(result, 99);
+//!     }
+//!
+//! If you want to call temporarily_*_foo from integration tests, you will have
+//! to get rid of the #[cfg(test)] attributes, and also add pub.
+
+use std::{cell::RefCell, thread::LocalKey};
+
+// This could be generic. That is, add a T parameter. Currently, only bool is supported, because YAGNI.
+#[must_use]
+pub struct Temporary {
+    flag: &'static LocalKey<RefCell<bool>>,
```

Comment:
I believe there is no reason to use `RefCell<T>` for any `T: Copy`, but it should be `Cell<T>`, which is simpler. It's similar to not using reference for passing arguments to a function if the argument is `Copy`

Reply:
My reason for RefCell<bool> is that we always do

```
thread_local! {
    static WIDGET: RefCell<Widget> = RefCell::new(...);
}
```

That is, thread_local + RefCell is a standard combination. AFAIK, we don't do anything else. Unless there is some big advantage of using Cell, I would stick with RefCell.

Reply:
> thread_local + RefCell is a standard combination

I'm challenging this. I think `thread_local` with either `Cell`/`RefCell` are equally valid. You might see `RefCell` more often just because more complex types appear more often. By not exposing an interface which allows borrowing, `Cell` is slightly [more efficient](https://stackoverflow.com/questions/72379106/what-are-the-differences-between-cell-refcell-and-unsafecell) since there is no need to check borrow rules in the runtime. In the IC monorepo, `Cell<u64>` appears just as frequent as `RefCell<u64>` within `thread_local`, although both are rare.

Reply:
I am not strongly against Cell -> Done.

I think the benefit of Cell is minimal though.

RefCell is definitely more common. Even the link you shared says so.